packages/gatsby-theme-docs/gatsby-node.js

```diff
@@ -122,15 +122,54 @@ exports.onCreateNode = ({ node, getNode, actions }, pluginOptions) => {
     path.resolve('src/releases')
   );
   if (isReleaseNotesPage) {
-    const releaseNoteSlug = trimTrailingSlash(slug) || '/';
+    let releaseNoteSlug = generateReleaseNoteSlug(
+      node.frontmatter.date,
+      node.frontmatter.title,
+      node.fileAbsolutePath
+    );
+    releaseNoteSlug = trimTrailingSlash(releaseNoteSlug) || '/';
     actions.createNodeField({
       node,
       name: 'slug',
-      value: `/releases${releaseNoteSlug}`,
+      value: releaseNoteSlug,
+    });
+    actions.createNodeField({
+      node,
+      name: 'date',
+      value: node.frontmatter.date,
+    });
+    actions.createNodeField({
+      node,
+      name: 'description',
+      value: node.frontmatter.description,
+    });
+    actions.createNodeField({
+      node,
+      name: 'type',
+      value: node.frontmatter.type,
+    });
+    actions.createNodeField({
+      node,
+      name: 'topics',
+      value: node.frontmatter.topics,
     });
   }
 };
 
+function generateReleaseNoteSlug(date = '', title = '', fileAbsolutePath) {
+  const basePath = '/releases';
+
+  if (fileAbsolutePath.endsWith('index.mdx')) {
+    return basePath;
+  }
+
+  const slug = `${date.split('T')[0]} ${title}`
+    .toLowerCase()
+    .replace(/[^a-z0-9]+/g, '-') // replace non alphanumeric with hyphen
+    .replace(/(^-|-$)+/g, ''); // remove pre or post string hyphens
+  return `${basePath}/${slug}`.replace(/\/\/+/g, '/'); // replace two or more slashes with single slash
```

Comment:
Can you explain a bit how the slug is supposed to be built? And I mean not just to me but add some comments in the code (for future references). Thanks

Reply:
After speaking with @nkuehn , it's better to use a library to be able to deal with things like transforming special characters and not miss information in the url. So I'm currently using [slugify](https://www.npmjs.com/package/slugify).